### PR TITLE
calculate innerTangents for 2 circles

### PR DESCRIPTION
**Pull request by raymondnijssen**
## Description
Adds the function innerTangents() to the QgsCircle class, which works quite similar to the already existing outerTangents() function. 

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by raymondnijssen** [comment]
image

File: src/core/geometry/qgscircle.h [hunk 1]
```diff
@@ -189,13 +189,38 @@ class CORE_EXPORT QgsCircle : public QgsEllipse
...
      */
     int outerTangents( const QgsCircle &other,
                        QgsPointXY &line1P1 SIP_OUT, QgsPointXY &line1P2 SIP_OUT,
                        QgsPointXY &line2P1 SIP_OUT, QgsPointXY &line2P2 SIP_OUT ) const;
 
+    /**
+     * Calculates the inner tangent points between this circle
+     * and an \a other circle.
+     *
+     * The outer tangent points correspond to the points at which the two lines
```
Comment (nyalldawson) [inline src/core/geometry/qgscircle.h]: "inner tangent"

File: src/core/geometry/qgscircle.h [hunk 2]
```diff
@@ -189,13 +189,38 @@ class CORE_EXPORT QgsCircle : public QgsEllipse
...
+     * which are drawn so that they are tangential to both circles but on
+     * different sides, touching the circles and crossing eachother.
```
Comment (nyalldawson) [inline src/core/geometry/qgscircle.h]: "each other"

File: src/core/geometry/qgsgeometryutils.h [hunk 3]
```diff
@@ -202,6 +202,29 @@ class CORE_EXPORT QgsGeometryUtils
       QgsPointXY &line1P1 SIP_OUT, QgsPointXY &line1P2 SIP_OUT,
       QgsPointXY &line2P1 SIP_OUT, QgsPointXY &line2P2 SIP_OUT );
 
+    /**
+     * Calculates the inner tangent points for two circles, centered at \a
+     * center1 and \a center2 and with radii of \a radius1 and \a radius2
+     * respectively.
+     *
+     * The outer tangent points correspond to the points at which the two lines
```
Comment (nyalldawson) [inline src/core/geometry/qgsgeometryutils.h]: Inner

**Review by lbartoletti — APPROVED** [review]

**Comment by lbartoletti** [comment]
You have to fix sip.
Thanks @raymondnijssen 
This will make it possible to propose a set of tools on the points of construction.

**Comment by raymondnijssen** [comment]
Thanks! Never thought my English would be worse than my cpp :D

**Pull request merged**